### DISCUSSION TITLE
Converting BigDecimal differs Scale

### DISCUSSION THREAD

**Discussion by acemint** [FasterXML/jackson-databind#4211] [Q&A]
I'm trying to convert a data with a generic Object as follow

```json
{"amount":100000000.23,"taxAmount":11000000.0}
``` 

But for some reason, the StringCache for the BigDecimal differs. One uses an exponent while the others don't. As a result, when using .toString to this class, it returns 
```json
{"amount":100000000.23,"taxAmount":1.1E7}
``` 

![image](https://github.com/FasterXML/jackson/assets/43749821/72bbd54e-a9cd-4350-a861-1843738fe4a8)

Any idea why this happens? Thanks

**Comment by cowtowncoder** [comment]
Please provide a bit of code to make sure we know what you mean. I am guessing "generic Object" means that you are doing something like:

```
Object value = objectMapper.readValue(json, Object.class);
```

but are you configuring mapper in any way? Enabling `DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS` maybe? (by default, FP values are read as `Double`s)

**Reply by acemint** [reply]
Yes I'm using your code above and I've enabled `DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS`. Both data types as shown have been converted as BigDecimal. However, the stringCache property for BigDecimal differs if the trailing decimal is 0

**Reply by acemint** [reply]
I guess I need to clarify my question:

I'd like to set the scale of the BigDecimal which is converted from the JsonDeserializer, because I do not want exponent symbol to appear in the BigDecimal, is there such a way to do that?

**Reply by cowtowncoder** [reply]
@Acemint There is no out-of-the-box option for post-processing `BigDecimal`. You may want to add custom deserializer to do that.

However: for JSON serialization side there is:

    JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN

enabling of which (applied via `JsonFactory`) will force non-scientific notation for `BigDecimal` values written out.
So depending on your exact needs that might work.

**Comment by acemint** [comment]
Found similar issue here
https://github.com/FasterXML/jackson-databind/issues/632

**Reply by acemint** [reply]
Reopened this discussion as with version 2.14.2 I still found the precision to be different according to the value passed, is it expected?

**Reply by cowtowncoder** [reply]
To be pedantic, I don't think precision is different, just notation. Actual value is not changed. I suspect it is JDK itself that changes representation, however (`BigDecimal` constructor)?

**Reply by cowtowncoder** [reply]
One thing to check is to see how basic `new BigDecimal("100000000.23")` (etc) works -- I think that will produce values you see, and Jackson doesn't really do anything special on read/deserialization side.
If so, I'd probably close this discussion: if there is something you would like Jackson to do, that'd be additional feature, but not anything that can be fixed per se (this may seem like a tiny distinction but it matters wrt version compatibility at least).
